### PR TITLE
Allow space or empty strings when using ros2 param set

### LINKED ISSUE #1092: Setting string parameter with empty content or space only content fails from ros2cli
## Bug report

**Required Info:**

- Operating System:
  - Ubuntu 22.03
- Installation type:
  - binaries
- Version or commit hash:
  - rolling (rclpy 3.7.1, but same code is in rolling branch at github)
- DDS implementation:
  -  Fast-RTPS
- Client library (if applicable):
  - rclpy

#### Steps to reproduce issue

* in one terminal start a node (ex using python3)
```
import rclpy.node
import rclpy
rclpy.init()
node = rclpy.node.Node("mynode", allow_undeclared_parameters=True, automatically_declare_parameters_from_overrides=True)
rclpy.spin(node)
```

* in a second terminal run one of the ros2cli command
```
ros2 param set /mynode myparam " "
ros2 param set /mynode myparam "  "
ros2 param set /mynode myparam ''
ros2 param set /mynode myparam ' '
ros2 param load param.yaml
```

with param.yaml
```
/mynode:
  ros__parameters:
    myparam: " "
```

#### Expected behavior
`Set parameter successful`
At minimum for the quoted space, the value should be a quoted space, which is a valid string character, but I expect a quoted empty string to be a valid string as well.

additionally, regardless of the exception in the current behaviour, I would expect the same behaviour as a standard string  (I added parenthesis to visualize empty and spaces) 
```
> ros2 param set /mynode myparam "foo"
Set parameter successful
> ros2 param get /mynode myparam
String value is:(foo)
```
for the space and empty space
```
> ros2 param set /mynode myparam " "
Set parameter successful
> ros2 param get /mynode myparam
String value is:( )

> ros2 param set /mynode myparam ""
Set parameter successful
> ros2 param get /mynode myparam
String value is:()
```

#### Actual behavior
```
Traceback (most recent call last):
  File "/opt/ros/rolling/bin/ros2", line 33, in <module>
    sys.exit(load_entry_point('ros2cli==0.21.0', 'console_scripts', 'ros2')())
  File "/opt/ros/rolling/lib/python3.10/site-packages/ros2cli/cli.py", line 89, in main
    rc = extension.main(parser=parser, args=args)
  File "/opt/ros/rolling/lib/python3.10/site-packages/ros2param/command/param.py", line 39, in main
    return extension.main(args=args)
  File "/opt/ros/rolling/lib/python3.10/site-packages/ros2param/verb/set.py", line 61, in main
    parameter.value = get_parameter_value(string_value=args.value)
  File "/opt/ros/rolling/local/lib/python3.10/dist-packages/rclpy/parameter.py", line 230, in get_parameter_value
    value.string_value = yaml_value
  File "/opt/ros/rolling/local/lib/python3.10/dist-packages/rcl_interfaces/msg/_parameter_value.py", line 239, in string_value
    assert \
AssertionError: The 'string_value' field must be of type 'str'
```

OR for the load
```
Traceback (most recent call last):
  File "/opt/ros/rolling/bin/ros2", line 33, in <module>
    sys.exit(load_entry_point('ros2cli==0.21.0', 'console_scripts', 'ros2')())
  File "/opt/ros/rolling/lib/python3.10/site-packages/ros2cli/cli.py", line 89, in main
    rc = extension.main(parser=parser, args=args)
  File "/opt/ros/rolling/lib/python3.10/site-packages/ros2param/command/param.py", line 39, in main
    return extension.main(args=args)
  File "/opt/ros/rolling/lib/python3.10/site-packages/ros2param/verb/load.py", line 53, in main
    load_parameter_file(node=node, node_name=node_name, parameter_file=args.parameter_file,
  File "/opt/ros/rolling/lib/python3.10/site-packages/ros2param/api/__init__.py", line 49, in load_parameter_file
    future = client.load_parameter_file(parameter_file, use_wildcard)
  File "/opt/ros/rolling/local/lib/python3.10/dist-packages/rclpy/parameter_client.py", line 307, in load_parameter_file
    param_dict = parameter_dict_from_yaml_file(parameter_file, use_wildcard)
  File "/opt/ros/rolling/local/lib/python3.10/dist-packages/rclpy/parameter.py", line 319, in parameter_dict_from_yaml_file
    return _unpack_parameter_dict(namespace, param_dict)
  File "/opt/ros/rolling/local/lib/python3.10/dist-packages/rclpy/parameter.py", line 341, in _unpack_parameter_dict
    parameter.value = get_parameter_value(str(param_value))
  File "/opt/ros/rolling/local/lib/python3.10/dist-packages/rclpy/parameter.py", line 230, in get_parameter_value
    value.string_value = yaml_value
  File "/opt/ros/rolling/local/lib/python3.10/dist-packages/rcl_interfaces/msg/_parameter_value.py", line 239, in string_value
    assert \
AssertionError: The 'string_value' field must be of type 'str'
```
#### Additional information

The culprit of this behavior is `yaml.safe_load()` that will generate a `NoneType` for "", '', " ", ' ' 
```
type(yaml.safe_load(" "))
<class 'NoneType'>
```
I can understand that strings must be passed through `yaml.safe_load`, but if the result is _NoneType_, it means that the `safe_load` did not generate a _ParseError_ or any other exception, and thus the _string_value_ should be used as is.
I tried that solution
`_value.string_value = yaml_value if yaml_value is not None else string_value`

which sets the value correctly, when the content is empty or space(s) 

( I must mention that set the parameter once does not set the parameter type, setting it twice does (for any `ros2 param set` call on allow_undeclared_parameters=True + automatically_declare_parameters_from_overrides=True )  but this is not related to that issue)

### PR DESCRIPTION
**Pull request by guihomework**
Should fix #1092

### CODE REVIEW

**Review by clalancette — CHANGES_REQUESTED** [review]
Generally looks like a good fix, two fixes needed to pass CI.

**Review by clalancette — APPROVED** [review]

**Comment by clalancette** [comment]
Oh, can you please sign off the latest commit and then force-push it?  That will make the DCO bot happy.

File: rclpy/test/test_parameter.py [hunk 1]
```diff
@@ -167,6 +168,48 @@ def test_double_array(self):
...
+                    type=int(ParameterType.PARAMETER_STRING_ARRAY),
+                    string_array_value=['foo', 'bar'])
+            ),
+        ]
+
+        for input_value, expected_value in test_cases:
+            try:
+                p = get_parameter_value(str(input_value))
+            except Exception as e:
+                assert False, f"failed to get param_value, reason: {e}"
```
[the remark this answers is not in the document]
Comment (guihomework) [reply rclpy/test/test_parameter.py]: @clalancette sorry, I had run my default pep8 and ignored all the line length issues but did not realize there were specific rules https://docs.ros.org/en/rolling/The-ROS2-Project/Contributing/Code-Style-Language-Versions.html#python. I am a newbie there, and don't even know yet how to enforce single quote from my editor. The pycodestyle pointed by the doc does not indicate the preferred style for strings, it must be defined somewhere else I suppose.

**Comment by clalancette** [comment]
CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/18241/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/12772/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/18913/)

**Review by fujitatomoya — COMMENTED** [review]
lgtm

**Pull request merged**